plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/FieldSetterFactory.java

```diff
@@ -351,53 +352,28 @@ public void setField(Block block, int position)
         }
     }
 
-    private static class TimestampFieldSetter
+    private class TimestampFieldSetter
             extends FieldSetter
     {
-        private final DateTimeZone timeZone;
         private final TimestampType type;
         private final TimestampWritableV2 value = new TimestampWritableV2();
 
-        public TimestampFieldSetter(SettableStructObjectInspector rowInspector, Object row, StructField field, TimestampType type, DateTimeZone timeZone)
+        public TimestampFieldSetter(SettableStructObjectInspector rowInspector, Object row, StructField field, TimestampType type)
         {
             super(rowInspector, row, field);
             this.type = requireNonNull(type, "type is null");
-            this.timeZone = requireNonNull(timeZone, "timeZone is null");
         }
 
         @Override
         public void setField(Block block, int position)
         {
-            long epochMicros;
-            int picosOfMicro;
-            if (type.isShort()) {
-                epochMicros = type.getLong(block, position);
-                picosOfMicro = 0;
+            Timestamp timestamp = getHiveTimestamp(type, block, position);
+            if (!ZoneOffset.UTC.equals(timeZone)) {
+                timestamp = convertTimestampToZone(timestamp, timeZone, ZoneOffset.UTC);
```

Comment:
Instead of importing hadoop function -- which is a foreign dependency and eg can get removed or have API changed in the future, and also operates on java.time, so it's a bit slower -- let's keep the code that we used to have here (`convertLocalEpochSecondsToUtc`).

Reply:
The previous version also had a foreign dependency, and it seems unlikely that the Hadoop utility is going to be changed (it hasn't been changed since it was added two years ago, and that method has its own tests).

It is _much_ slower though,<sup>1</sup> because it converts the timestamp to a string and re-parses it as a `TimestampTZ` before doing the conversion. I'll switch it back.

<sup>1</sup> About 300 times slower than using java.time directly.